src/System.Data.SqlClient/tests/ManualTests/SQL/AsyncTest/AsyncTest.cs

```diff
@@ -91,5 +92,99 @@ public static void FailureTest()
             readerTask.Result.Dispose();
             con.Close();
         }
+
+
+        [CheckConnStrSetupFact]
+        public static void TestReadAsync()
+        {
+            CompareReadSyncAndReadAsync();
+        }
+
+        private static async void CompareReadSyncAndReadAsync()
+        {
+            const string sql = "SET NOCOUNT ON"
+                            + " SELECT 'a'"
+                            + " DECLARE @t DATETIME = SYSDATETIME()"
+                            + " WHILE DATEDIFF(s, @t, SYSDATETIME()) < 20 BEGIN"
+                            + "   SELECT 2 x INTO #y"
+                            + "   DROP TABLE #y"
+                            + " END"
+                            + " SELECT 'b'";
+            double elapsedAsync = await RunReadAsync(sql);
+            double elapsedSync = RunReadSync(sql);
+            Assert.True(elapsedAsync < (elapsedSync / 2.0d));
```

Comment:
Do you plan to improve this assert in future commits?

Reply:
@saurabh500 
Yes. I am not sure what would be the most appropriate criteria for passing the test.
One approach I am considering would be : 
```
Assert.True(elapsedAsync < 50) // ReadAsync() finished within 50 milliseconds
```
What do you think?